src/adapter/InMemoryRequestRepositoryAdapter.js

```diff
@@ -0,0 +1,27 @@
+const RequestRepositoryPort = require('../domain/request/port/RequestRepositoryPort');
+const RequestStateEnum = require('../domain/request/RequestStateEnum');
+
+class InMemoryRequestRepositoryAdapter extends RequestRepositoryPort {
+  constructor() {
+    super();
+    this.requests = new Map();
+  }
+
+  save(request) {
+    this.requests.set(request.id, request);
+  }
+
+  getScheduledRequestsWithValidFromBeforeNow() {
+    return Array.from(this.requests.values()).filter(
```

Comment:
Array.from(this.requests.values())
.filter(request => request.state.name === RequestStateEnum.SCHEDULED)
.filter(request =>request.validFrom < Date.now()) is more readable